rust/etw_helpers/src/sessions.rs

```diff
@@ -0,0 +1,370 @@
+use std::{borrow::Cow, mem::ManuallyDrop};
+
+use windows::{core::PCSTR, Win32::System::Diagnostics::Etw::*};
+
+#[repr(C)]
+pub struct EventTraceProperties {
+    props: EVENT_TRACE_PROPERTIES,
+    session_name: [u8; 1024],
+    file_name: [u8; 1024],
+}
+
+impl EventTraceProperties {
+    fn empty_for_query() -> EventTraceProperties {
+        unsafe {
+            let mut props: EventTraceProperties = core::mem::zeroed();
+            props.props.Wnode.BufferSize = core::mem::size_of::<Self>() as u32;
+            props.props.Wnode.Flags = WNODE_FLAG_TRACED_GUID;
+
+            props.props.LoggerNameOffset = core::mem::size_of::<EVENT_TRACE_PROPERTIES>() as u32;
+            props.props.LogFileNameOffset =
+                core::mem::size_of::<EVENT_TRACE_PROPERTIES>() as u32 + 1024;
+
+            props
+        }
+    }
+
+    pub fn set_session_name(&mut self, session_name: &str) -> &mut Self {
+        if !session_name.is_empty() {
+            unsafe {
+                let len = session_name.len();
+                if len >= 1024 {
+                    panic!()
+                }
+
+                // TODO: Can we actually use UTF-8 as a session name?
```

Comment:
Short answer is no. So the current library works fine for ASCII but breaks for anything non-ASCII.

If we're putting this library out for public use, we should use the UTF16 APIs, even though it's more work.